### ISSUE TITLE
[Issue] Make Dmods interact with Rugged Construction hullmod

### ISSUE THREAD

**Issue by RkShaRkz** [RkShaRkz/Starsector-IntelliJ-Template#62]
### Mod Name:
VayraMerged

### Mod Version:
All of them

### What did you expect to happen?
Alot of dmods have their effects 'halved' when Rugged Construction is present; none of vayramerged ones have that

### What actually happened?
inappilicable, and they all just work as they did before rugged construction as a thing.

### Steps to Reproduce:

1. 
2. 
3. 

### Additional info:
> Hound, for example, has Rugged Construction. But Malfunctioning Ammo Forge makes anything installed on it do fragmentation. Which is like turning an Assault Chaingun into...a vulcan cannon, basically.

So this Hound has it; make these Dmods try to affect it 50% less, or have their effects halved somehow when rugged construction is present and that should be it i think.

**Comment by RkShaRkz** [comment]
The rugged construction's description is 
```
Reduces most negative effects of d-mods by %s. If disabled or destroyed in combat, the ship has a %s chance to avoid new d-mods, and is almost always recoverable after the battle.

In addition, the supply cost to recover from deployment - and to effect repairs if the ship is disabled - is reduced by %s.
```

The rugged construction's id is `rugged`

too bad i can't see this shit, or the source, but... I need to halve all of it's bad stuff. So do some 50%-ish of everything if `rugged` is present, built-in or Smodded builtin.

**Comment by RkShaRkz** [comment]
Merged-in in commit c04d330

**Comment by RkShaRkz** [comment]
Marking as done and resolved.
